in_session.go

```diff
@@ -27,6 +27,16 @@ type inSession struct{ loggedOn }
 func (state inSession) String() string { return "In Session" }
 
 func (state inSession) FixMsgIn(session *session, msg *Message) sessionState {
+
+	if session.CleanIncomingHotPath && IsExecutionReport(msg.rawMessage.Bytes()) {
+		if err := session.application.FromApp(msg, session.sessionID); err != nil {
+			return handleStateError(session, err)
+		}
+		if err := session.store.IncrNextTargetMsgSeqNum(); err != nil {
+			return handleStateError(session, err)
+		}
+		return state
+	}
```

Comment:
I don't think we need this check here; in the below switch, it's already checking for the message type. The main changes can be at line 64 below

Reply:
On line 40 we do https://github.com/alpacahq/quickfix/blob/75f208b87f5f26139db3883716f399ee31c888cb/in_session.go#L40-L40 which is using 
https://github.com/alpacahq/quickfix/blob/75f208b87f5f26139db3883716f399ee31c888cb/field_map.go#L123-L123
But tagLookup is being initialised inside `ParseMessageWithDataDictionary`
https://github.com/alpacahq/quickfix/blob/75f208b87f5f26139db3883716f399ee31c888cb/message.go#L227-L228

So I would avoid removing this check. What do you think?